realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -165,8 +166,12 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_Table_nativeIsColumnNullable(J
         ThrowException(env, UnsupportedOperation, "Not allowed to convert field in subtable.");
         return JNI_FALSE;
     }
+
     size_t column_index = S(columnIndex);
-    return to_jbool(table->is_nullable(column_index));
+    if (table->get_column_type(S(columnIndex)) != type_Table) {
+        return to_jbool(table->is_nullable(column_index)); // noexcept
+    }
+    return to_jbool(table->get_descriptor()->get_subdescriptor(S(columnIndex))->is_nullable(S(0))); // noexcept
```

Comment:
why ? would not primitive list be always nullable?

Reply:
no.

*nullable* for primitive list means that the list can contain null as value.

Reply:
fixed